### PR TITLE
fix(appsync): add caching config to AppSync resolvers

### PR DESCRIPTION
**Pull request by kylevillegas93**
While trying to add caching config to some of my application's resolvers, I discovered that the BaseResolverProps do not include caching configuration like the CfnResolver does.

This PR adds this missing caching configuration to the BaseResolverProps and adds the configuration as part of the creation of the CfnResolver. 

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by GarrettClyde** [comment]
Awesome.  Thanks for doing this!  Looking forward to its release.

File: packages/@aws-cdk/aws-appsync/lib/caching-config.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import { Duration } from '@aws-cdk/core';
+
+/**
+ * CachingConfig for AppSync resolvers
+ */
+export interface CachingConfig {
+  /**
+   * The caching keys for a resolver that has caching enabled.
+   * Valid values are entries from the $context.arguments, $context.source, and $context.identity maps.
```
Comment (otaviomacedo) [inline packages/@aws-cdk/aws-appsync/lib/caching-config.ts:9]: Can you please also add validation for this constraint?

File: packages/@aws-cdk/aws-appsync/lib/caching-config.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+   *
+   * @default - No caching keys
+   */
+  readonly cachingKeys?: string[];
+
+  /**
+   * The TTL in seconds for a resolver that has caching enabled.
+   * Valid values are between 1 and 3600 seconds.
```
Comment (otaviomacedo) [inline packages/@aws-cdk/aws-appsync/lib/caching-config.ts:17]: Same thing with this one.

File: packages/@aws-cdk/aws-appsync/lib/caching-config.ts (hunk 1 above)
Comment (kylevillegas93) [reply packages/@aws-cdk/aws-appsync/lib/caching-config.ts:9]: Will do!

File: packages/@aws-cdk/aws-appsync/lib/caching-config.ts (hunk 2 above)
Comment (kylevillegas93) [reply packages/@aws-cdk/aws-appsync/lib/caching-config.ts:17]: Will do!

File: packages/@aws-cdk/aws-appsync/lib/caching-config.ts (hunk 1 above)
Comment (kylevillegas93) [reply packages/@aws-cdk/aws-appsync/lib/caching-config.ts:9]: Alright - added validation for caching keys - added some constants for $context.source, $context.arguments and $context.identity to help validate.

File: packages/@aws-cdk/aws-appsync/lib/caching-config.ts (hunk 2 above)
Comment (kylevillegas93) [reply packages/@aws-cdk/aws-appsync/lib/caching-config.ts:17]: Added validation + unit tests for ttl!

File: packages/@aws-cdk/aws-appsync/lib/caching-config.ts (hunk 1 above)
Comment (otaviomacedo) [reply packages/@aws-cdk/aws-appsync/lib/caching-config.ts:9]: Thanks! One last thing: Before validating you should check whether the value is [encoded as a token][token], using the `Token.isUnresolved()` static method.

[token]: https://docs.aws.amazon.com/cdk/latest/guide/tokens.html#tokens_encoding

Comment (kylevillegas93) [reply packages/@aws-cdk/aws-appsync/lib/caching-config.ts:9]: Ah cool - I'll make this change!

Comment (kylevillegas93) [reply packages/@aws-cdk/aws-appsync/lib/caching-config.ts:9]: Alright sweet - done! Checked to see if the caching key is resolved first before validating.

**Review by otaviomacedo — APPROVED** [review]

**Review by otaviomacedo — APPROVED** [review]

**Pull request merged**